docs/virtual_machines/startup_scripts.md

```diff
@@ -112,6 +113,17 @@ ConfigDrive data source:
         cloudInitConfigDrive:
           userData: "#cloud-config"
 
+When using the ConfigDrive datasource, the `networkData` part has to be in the
+[OpenStack Metadata Service Network](https://specs.openstack.org/openstack/nova-specs/specs/liberty/implemented/metadata-service-network-info.html) format:
+```
+       volumes:
+      - name: cloudinitvolume
+        cloudInitConfigDrive:
+          networkData: |
+            {"links":[{"id":"enp2s0","type":"phy","ethernet_mac_address":"02:26:19:00:00:30"}],"networks":[{"id":"NAD1","type":"ipv4","link":"enp2s0","ip_address":"10.184.0.244","netmask":"255.255.240.0","routes":[{"network":"0.0.0.0","netmask":"0.0.0.0","gateway":"23.253.157.1"}],"network_id":""}],"services":[]}
```

Comment:
For a minimal example you may drop the empty `network_id` and the empty `services` list.

Reply:
The reason I kept `network_id` and `services` is that this format has a [JSON schema](https://docs.openstack.org/nova/latest/_downloads/9119ca7ac90aa2990e762c08baea3a36/network_data.json) to validate the data.

I think we should stick to the format, even though cloud-init is more [permissive](https://github.com/canonical/cloud-init/blob/7ffd0bcd8379ae6ece096e7014444978efa5f7c8/cloudinit/sources/helpers/openstack.py#L526).

Reply:
ACK, fair point.